assessment.rb

```diff
@@ -1,12 +1,67 @@
 require 'csv'
 # Assessment answers to:
 
+class CreateLicensees < ActiveRecord::Migration
+  def change
+    create_table :licensees do |t|
+      t.string :surname
+      t.string :given_name
+      t.string :gender
+      t.integer :height
+      t.integer :weight
+      t.date :dob
+      t.string :eye_color
+      t.string :hair_color
+      t.timestamps null: false
+    end
+  end
+end
+
 # question 2:
 
+licensee = Licensee.create surname: 'Bishop', given_name: 'Caren', gender: 'f', height: 63, weight: 132, dob: '1943-09-26', eye_color: 'Brown', hair_color: 'Black'
+p licensee
+
 # question 3:
 
+namespace :db do
+  namespace :populate do
+    desc 'Fill the licensees table with example data'
+    task licensees: :environment do
+      Licensee.transaction do
+        CSV.foreach(Rails.root + 'data/licensees.csv',
+                    headers: true) do | licensee_row |
+          licensee = licensee_row.to_hash
+          next if Licensee.exists? licensee
+          Licensee.create! licensee
+        end
+      end
+    end
+  end
+end
+
 # question 4:
 
+Licensee.all.select do |l|
+  l.eye_color == 'Hazel'
+  and l.hair_color == 'Brown'
+    or l.hair_color == 'Black'
+end
+
```

Comment:
Actually, let me amend what I said - in Ruby, logical composition operators (`and`, `or`) are of equal precedence to each other (as opposed to `&&` or `||`; `&&` has higher precedence than `||`), so it will default to reading left-to-right. As a result, what you're really saying in this condition is:
`(l.eye_color == 'Hazel' and l.hair_color == 'Brown') or l.hair_color == 'Black'`

This is pretty obscure, but it's also a good reason to favor `&&` and `||` over `and` and `or` - their behavior is more predictable.

http://www.techotopia.com/index.php/Ruby_Operator_Precedence